src/Pages/MovieListPage/MovieListPage.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect, useRef, useCallback } from "react";
+import { SearchForm } from '../../Components/SearchForm/SearchForm';
+import { GenreSelector } from '../../Components/GenreSelector/GenreSelector';
+import { SortControl } from '../../Components/SortControl/SortControl';
+import { MovieTile } from '../../Components/MovieTileComponent/MovieTile';
+import { MovieDetails } from '../../Components/MovieDetails/MovieDetails';
+import { Header } from '../../Components/Header/Header';
+import './MovieListPage.css';
+
+const genreList = [
+    { id: 0, name: 'all' },
+    { id: 1, name: 'drama' },
+    { id: 2, name: 'horror' },
+    { id: 3, name: 'comedy' },
+    { id: 4, name: 'documentary' },
+    { id: 5, name: 'crime' },
+];
+
+const sortOptions = [
+    {
+        id: 1,
+        desc: 'Release Date'
+    },
+    {
+        id: 2,
+        desc: 'Title'
+    }
+];
+
+
+export function MovieListPage(props) {
+
+    const [searchValue, setSearchValue] = useState('');
+    const [selectedGenre, setSelectedGenre] = useState(0);
+    const [sortValue, setSortValue] = useState(1);
+    const [selectedMovieTile, setselectedMovieTile] = useState(null);
+    const [movieList, setMovieList] = useState([]);
+    const [consumingFetch, setConsumingFetch] = useState(false);
+    const [controller] = useState(new AbortController());
+    const [queryParams] = useState(new URLSearchParams());
+    const defaultSearch = 'Jumanji';
+
+    const onSearch = function (input) {
+        queryParams.set('searchBy', 'title');
+        queryParams.set('search', input);
+        console.log(input)
+        setSearchValue(input);
+    }
+
+    const onSelectGenre = function (genre) {
+        setSelectedGenre(genre);
+        if (genre.id === 0) {
+            queryParams.delete('searchBy');
+            queryParams.delete('search');
+        } else {
+            queryParams.set('searchBy', 'genres');
+            queryParams.set('search', genre.name);
+        }
+    }
+
+    const onSort = function (sortValue) {
+        console.log('sortValue :>> ', sortValue);  
+        queryParams.set('sortBy', sortValue === '2' ? 'title' : 'release_date');
+        setSortValue(sortValue);
+    }
+
+    const onSelectMovieTile = function (movie) {
+        setselectedMovieTile(movie);
+    }
+
+    const searchMovies = useCallback(async () => {
```

Comment:
Abort might not work as expected because the aborted controller is used to the new request. You might need to initialize a new abort controller. Try something like this and test with network throttle:
```
const searchMovies = useCallback(async () => {
        consumingFetch && controller.abort();

        setConsumingFetch(true);

        const newController = new AbortController();
        setController(newController);
        const signal = newController.signal;

        //create queryparams
        queryParams.set('sortOrder', 'asc');

        signal.addEventListener("abort", () => {
            console.log("aborted!")
        });

        try {
            const response = await fetch("http://localhost:4000/movies?" + queryParams.toString(), {
                signal,
            });
            const movies = await response.json();

            let moviesmapped = movies.data.map(m => {
                let movie = {};
                movie.urlImage = m.poster_path;
                movie.name = m.title;
                movie.releaseYear = m.release_date;
                movie.genres = m.genres;
                movie.rating = m.vote_average;
                movie.duration = m.runtime;
                movie.description = m.overview;
                movie.id = m.id;
                return movie;
            });

            setConsumingFetch(false);
            setMovieList(moviesmapped);
        } catch (e) {
            console.error({e});
        }
    });
```